optuna/integration/botorch.py

```diff
@@ -400,14 +402,21 @@ def __init__(
         constraints_func: Optional[Callable[[FrozenTrial], Sequence[float]]] = None,
         n_startup_trials: int = 10,
         independent_sampler: Optional[BaseSampler] = None,
+        seed: Optional[int] = None,
     ):
         _imports.check()
 
         self._candidates_func = candidates_func
         self._constraints_func = constraints_func
-        self._independent_sampler = independent_sampler or RandomSampler()
+        self._independent_sampler = independent_sampler or RandomSampler(seed=seed)
         self._n_startup_trials = n_startup_trials
 
+        if seed is not None:
+            # `torch.manual_seed` makes `BoTorchSampler` and the default candidates functions
+            # reproducible. `SobolQMCNormalSampler`'s constructor has a `seed` argument, but its
+            # behavior is deterministic when the torch's seed is fixed.
+            torch.manual_seed(seed)
```

Comment:
By using `torch.manual_seed`, is there something side effect that change the behavior of user program unexpectedly?

Reply:
I changed to use BoTorch's `manual_seed`, which is a context manager. PTAL.